src/components/table-row.tsx

```diff
@@ -46,7 +52,21 @@ const TableRow = ({
     maxHeightStyle: defaultMaxHeightStyle
   })
 
+  /**
+   * state variable to open and close delete confirmation modal window
+   */
+  const [showDeleteConfirmationModal, setShowDeleteConfirmationModal] = useState<any>({
```

Comment:
Add proper type for it and also `show` is not needed. You should just set this variable to null if you want to close the modal. And as long as it has a not-null value, you will show the modal.

```ts
const [showDeleteConfirmationModal, setShowDeleteConfirmationModal] = useState<{
  onConfirm: () => void,
  buttonLabel: string
} | null>(null);
```